Tests/UnitTests/Core/Detector/BlueprintTests.cpp

```diff
@@ -0,0 +1,66 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Detector/Blueprint.hpp"
+
+#include <fstream>
+
+namespace Acts {
+namespace Experimental {
+class IInternalStructureBuilder {};
+}  // namespace Experimental
+}  // namespace Acts
+
+BOOST_AUTO_TEST_SUITE(Experimental)
+
+BOOST_AUTO_TEST_CASE(BlueprintTest) {
+  std::vector<Acts::ActsScalar> bValues = {0., 10., 100.};
+
+  // Create  root node
+  std::vector<Acts::BinningValue> binning = {Acts::binR};
+  auto root = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "detector", Acts::Transform3::Identity(), Acts::VolumeBounds::eOther,
+      bValues, binning);
+  // Check the root node
+  BOOST_CHECK(root->isRoot());
```

Comment:
Maybe you could add a couple of these checks again, after you added new branches?

Reply:
Added tests to cover the entire parent/child phase space.